clang/lib/CodeGen/CGObjCGNU.cpp

```diff
@@ -2222,6 +2222,96 @@ class CGObjCObjFW: public CGObjCGNU {
     return ClassSymbol;
   }
 
+  void GenerateDirectMethodPrologue(
+      CodeGenFunction &CGF, llvm::Function *Fn, const ObjCMethodDecl *OMD,
+      const ObjCContainerDecl *CD) override {
+    auto &Builder = CGF.Builder;
+    bool ReceiverCanBeNull = true;
+    auto selfAddr = CGF.GetAddrOfLocalVar(OMD->getSelfDecl());
+    auto selfValue = Builder.CreateLoad(selfAddr);
+
+    // Generate:
+    //
+    // /* for class methods only to force class lazy initialization */
+    // self = [self self];
+    //
+    // /* unless the receiver is never NULL */
+    // if (self == nil) {
+    //     return (ReturnType){ };
+    // }
+    //
+    // _cmd = @selector(...)
+    // ...
+
+    if (OMD->isClassMethod()) {
+      const ObjCInterfaceDecl *OID = cast<ObjCInterfaceDecl>(CD);
+      assert(
+          OID &&
+          "GenerateDirectMethod() should be called with the Class Interface");
```

Comment:
`cast` is already an asserting cast, so this should just be a comment. But really, if this is reliably true (it's never passed an implementation or a category?), I wonder why it's not expressed in the type of this method.

Reply:
This is straight up copied from CGObjCMac, so I have no idea why it was done this way :)

Reply:
Heh, alright.